### PR TITLE
add support for adg2128

### PR DESCRIPTION
**Pull request by buha**
ADG2128 is an 12x8 cross point switch. As it stands, IIO does not
have any support for cross point switches. So in order to have a
way to drive this chip using libiio, this commit adds a driver
implementation for adg2128 written in python that makes use of the
direct register access feature of IIO to perform I2C read/write
operations.

Ideally, IIO should be extended to support such devices (cross
point switches) but until this happens, this seems to be an
acceptable compromise to be able to communicate with adg2128
device in the same IIO connection with other ADI devices.

examples/adg2128.py shows how the driver exposes a 12x8 2D
array-like access to the switches to the user:
device[x][y] = True/False # closed/open

The driver supports communicating with several adg2128 devices
on the same bus, with common Y connections and all this is
presented to the user in a larger 2D array:
- 24x8 for 2 devices
- 36x8 for 3 devices
...

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] examples/adg2128.py

**Test Configuration**:
* Hardware: SDP-K1 board + (undisclosed) board featuring 2x ADG2128
* OS: no-OS running an iio server exposing a basic adg2128 iio device which only features debug register access interface

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: adi/rx_tx.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def rx_enabled_channels(self, value: Union[List[int], List[str]]):
         translated to channel numbers.
         """
         if not value:
-            raise Exception("rx_enabled_channels cannot be empty")
+            return
```
Comment (adrimbarean) [inline adi/rx_tx.py:139]: This is just an idea, but should we consider still displaying a warning here, without raising an exception? Something along the lines of `rx_enabled_channels empty; only debug interface is available`?

**Review by adrimbarean — COMMENTED** [review]
Looks good to me. There are some static analysis issues mentioned that I think also stop the doc build. Fixing them should pass the automatic builds.

**Comment by tfcollins** [comment]
Can you add a doc page for the new device class?

File: adi/rx_tx.py (hunk 1 above)
Comment (buha) [reply adi/rx_tx.py]: yeah, I just re-implemented this with attribute class inheritance (no rx stuff) as per @tfcollins suggestion

**Comment by buha** [comment]
I've also added those .rst files changes to newest version.
(Oddly enough, running `python update_devs.py` edits lots of .rst files, not sure if this is normal.)

**Comment by buha** [comment]
@tfcollins @adrimbarean any more feedback on this ? I tried to solve all Lint/Codacy errors but they kinda go against each other, fixing one breaks the other and viceversa

**Review by adrimbarean — APPROVED** [review]

**Comment by buha** [comment]
@tfcollins any more feedback ?

**Comment by tfcollins** [comment]
Please add a basic test to test the different switch combinations and run the linter pre-commit pieces

**Comment by tfcollins** [comment]
```
codespell................................................................Passed
isort....................................................................Failed
- hook id: isort
- files were modified by this hook
Skipped 2 files
Fixing /home/runner/work/pyadi-iio/pyadi-iio/adi/adg2128.py
Skipped 1 files
Check Yaml...............................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Debug Statements (Python)................................................Passed
Check docstring is first.................................................Passed
Flake8...................................................................Passed
black....................................................................Passed
blacken-docs.............................................................Passed
```

Some of the imports are out of order

**Comment by buha** [comment]
ok, just pushed the isort fix

I use a system that comes with python3.10 and pre-commit crashes with library errors for isort with python3.10 
To make it work I had to use virtualenv with 3.9 for isort to work

will also add a test

**Comment by tfcollins** [comment]
> ok, just pushed the isort fix
> 
> I use a system that comes with python3.10 and pre-commit crashes with library errors for isort with python3.10 To make it work I had to use virtualenv with 3.9 for isort to work
> 
> will also add a test

If you can paste the error I can probably help. I did fix one of these issues a few months bacl

**Comment by buha** [comment]
- added test
- regarding codacy failure, i think it is impossible to fix all those codacy complaints because fixing them breaks other tests (doc or lint)

**Comment by tfcollins** [comment]
@buha please rebase on master

**Comment by buha** [comment]
rebased on master

**Comment by tfcollins** [comment]
@buha pre-commit is still failing. Rebase got you the py3.10 fix but __init__ changed a bit. Run pre-commit again please

**Comment by buha** [comment]
@tfcollins rebased again, fixed pre-commit

**Pull request merged**